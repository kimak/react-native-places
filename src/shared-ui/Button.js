import React from "react";
import { Text, TouchableOpacity, StyleSheet } from "react-native";

const styles = StyleSheet.create({
  button: {
    padding: 5,
    marginTop: 10,
    borderWidth: 1,
    borderColor: "#CCC",
    alignSelf: "flex-start"
  }
});

const AppButton = props => (
  <TouchableOpacity onPress={props.onPress} style={styles.button}>
    <Text>{props.label}</Text>
  </TouchableOpacity>
);

export default AppButton;
