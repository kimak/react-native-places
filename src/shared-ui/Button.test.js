import React from "react";
import Button from "./Button";
import renderer from "react-test-renderer";

describe("Button", () => {
  it("Button renders correctly with label", () => {
    const tree = renderer.create(<Button label="OK" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Button renders correctly without label", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
