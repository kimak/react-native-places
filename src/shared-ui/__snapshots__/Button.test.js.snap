// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button renders correctly with label 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "borderColor": "#CCC",
      "borderWidth": 1,
      "marginTop": 10,
      "opacity": 1,
      "padding": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    OK
  </Text>
</View>
`;

exports[`Button Button renders correctly without label 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "borderColor": "#CCC",
      "borderWidth": 1,
      "marginTop": 10,
      "opacity": 1,
      "padding": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  />
</View>
`;
